src/host.rs

```diff
@@ -146,35 +148,40 @@ impl<'a> RuntimeArgs<'a> {
 /// }
 /// ```
 pub trait Externals {
+    /// Type of error the host can yield while executing a host function.
+    type Error: CanResume + From<Error>;
```

Comment:
You might also want to have `Into<Error>` so that we kinda guarantee that we can carve out a `wasmi::Error` (trap) from a user error.

Reply:
Unless I am missing something, [it look like the current code is very permissive as any host error make the interpreter resumable?](https://github.com/paritytech/wasmi/pull/384/files#diff-68c6f449503a33c74a15800c0a9c8fd34028465a06edcac92b32bbb9db64bbceL328-R330)

Reply:
For the `Into<Error>`, it's impossible as the host error is variant containing the trap, i.e. you can inject the trap but the projection is not always possible. We could have a `TryInto` but it's not used anywhere in the code right now? Did I understood what you meant?

Reply:
Yeah I see so the best we can have is an optional mapping from `HostError` to `wasmi::Error`, like `Into<Option<Error>>`. `TryInto` is not really good because we actually do not need an error variant there imo. `Into<Option<Error>>` looks very verbose on the other hand.